finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/Message.scala

```diff
@@ -60,7 +61,8 @@ object Headers {
     def toSeq = synchronized(current)
     override def toString = s"""Headers(${toSeq.mkString(", ")})"""
     def contains(key: String) = toSeq.exists { case (k, _) => key == k }
-    def get(key: String) = toSeq.collect { case (k, v) if key == k => v }
+    def get(key: String) = toSeq.find(_._1 == key).map(_._2)
+    def getAll(key: String) = toSeq.filter(_._1 == key).map(_._2)
```

Comment:
I removed it per [alex's comment](https://github.com/linkerd/linkerd/pull/1298#discussion_r116559219), but easy enough to revert this. I could also put together a benchmark to see which is faster.